make tests: upon db connect: SCRAM authentication requires libpq version 10 or above

@alistaire47 saw

```
/usr/local/lib/python3.11/site-packages/psycopg2/__init__.py:122: in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
E   sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) SCRAM authentication requires libpq version 10 or above
```
during
```
conbench/db.py:128: in create_all
    Base.metadata.create_all(engine)
```
As of the nature of the error (permanent), the retry loop is of course not resolving this.

What's a little puzzling about this error is that it suggests that we use different versions of libpq across people/machines; but as of our containerized workflow we try to prevent exactly this. The error is probably a red herring.

Btw, we pip-install https://pypi.org/project/psycopg2-binary/ in the app's container image, and that bundles a pre-built version of libpq.

--- a/Dockerfile
+++ b/Dockerfile
@@ -19,7 +19,7 @@ FROM python:3.11-slim
 # curl is needed for docker-compose health checks. `git` is needed by some unit
 # tests as of today.
 RUN apt-get update && apt-get install -y -q --no-install-recommends \
-    curl git build-essential && apt-get clean && rm -rf /var/lib/apt/lists/*
+    curl git build-essential libpq-dev && apt-get clean && rm -rf /var/lib/apt/lists/*
 
 COPY requirements-webapp.txt /tmp/
 COPY requirements-dev.txt /tmp/
@@ -72,3 +72,4 @@ COPY ./buildinfo.json /buildinfo.json
 # Re-active this to get ideas for how the image size can be further reduced.
 #RUN echo "biggest dirs"
 #RUN cd / && du -ha . | sort -r -h | head -n 50 || true
+

--- a/conbench/db.py
+++ b/conbench/db.py
@@ -1,12 +1,11 @@
 import functools
 import logging
 
+import psycopg2
 import sqlalchemy.exc
 import tenacity
-import psycopg2
 from sqlalchemy import create_engine
 from sqlalchemy.orm import scoped_session, sessionmaker
-
 
 from .config import Config
 

--- a/conbench/db.py
+++ b/conbench/db.py
@@ -3,8 +3,10 @@ import logging
 
 import sqlalchemy.exc
 import tenacity
+import psycopg2
 from sqlalchemy import create_engine
 from sqlalchemy.orm import scoped_session, sessionmaker
+
 
 from .config import Config
 
@@ -14,6 +16,9 @@ Session = scoped_session(session_maker)
 
 
 log = logging.getLogger(__name__)
+
+
+log.info("psycopg2.__libpq_version__: %s", psycopg2.__libpq_version__)
 
 
 def configure_engine(url):

--- a/requirements-webapp.txt
+++ b/requirements-webapp.txt
@@ -20,7 +20,7 @@ pandas
 prometheus-client
 prometheus-flask-exporter
 psutil
-psycopg2-binary
+psycopg2
 py-cpuinfo
 pytest>=7.0.0
 python-dotenv

